### PR TITLE
Add a Bridge Device type

### PR DESCRIPTION
**Pull request by arunbharadwaj**
#### Problem
We need a Bridge Device type as per CHIP Spec

 #### Summary of Changes
- Add a new Bridge Device type to chip-devices.xml
- Enable Basic cluster in Bridge Device
- Placeholder for Descriptor cluster in the Bridge Device to be enabled after a separate PR for that lands
- Edit bridge-app.zap to add Bridge Device at Endpoint 0
- Modify CHIP_ZCL_ENDPOINT_MIN to 0 such that Bridge Device can be reported via chip-tool

### CODE REVIEW

**Review by jrhees-cae — COMMENTED** [review]
Do additions need to be made to the chip-tool example to be able to read these attributes on EP0?

**Comment by arunbharadwaj** [comment]
> Do additions need to be made to the chip-tool example to be able to read these attributes on EP0?

There was one change required - had to change CHIP_ZCL_ENDPOINT_MIN to 0 instead of 1. I have that patch as a part of this series as well. Other than that nothing else is needed since I have only enabled Basic cluster on EP0. After the Descriptor cluster definition lands, we need more changes to chip-tool to read those attributes.

File: examples/bridge-app/bridge-common/bridge-app.zap
Comment (chrisdecenzo) [inline examples/bridge-app/bridge-common/bridge-app.zap:5293]: are you able to specify explicitly an endpoint id or does it just compute based upon the array index?

**Review by chrisdecenzo — APPROVED** [review]
Looks harmless as an incremental update but looking forward to seeing Descriptor cluster!

File: examples/bridge-app/bridge-common/bridge-app.zap
Comment (arunbharadwaj) [reply examples/bridge-app/bridge-common/bridge-app.zap:5293]: > are you able to specify explicitly an endpoint id or does it just compute based upon the array index?

We are able to specify the `endpointId` explicitly in the zap tool. There is also a `endpointTypeIndex` in the zap file which gets generated automatically.

**Review by msandstedt — APPROVED** [review]

File: examples/bridge-app/linux/main.cpp
Comment (vivien-apple) [inline examples/bridge-app/linux/main.cpp:77]: I have added `MfgSpecificPing` command mostly to be able our old `echo` mechanism and in order to check that manufacturer specific methods on a cluster are generated properly. If you don't need it you can just disable the `MfgSpecificPing` method from the basic cluster in the ZAP tool.

It depends if you really want it or not. For the record this method is just a test to send a message to the device, and it will reply back with a default response that contains `EMBER_ZCL_STATUS_SUCCESS`.

File: src/app/zap-templates/zcl/chip-devices.xml
Comment (vivien-apple) [inline src/app/zap-templates/zcl/chip-devices.xml:211]: Please remove the commented lines. We will add them back later when needed.

**Review by vivien-apple — APPROVED** [review]
Approving assuming the commented block in `chip-devices.xml` will be removed before merging.

File: src/app/zap-templates/zcl/chip-devices.xml
Comment (arunbharadwaj) [reply src/app/zap-templates/zcl/chip-devices.xml]: > Please remove the commented lines. We will add them back later when needed.
Done

File: examples/bridge-app/linux/main.cpp
Comment (arunbharadwaj) [reply examples/bridge-app/linux/main.cpp]: Done

**Review by jrhees-cae — APPROVED** [review]
LGTM as an incremental change

**Review by woody-apple — APPROVED** [review]

**Pull request merged**
